.github/workflows/labeler.yml

```diff
@@ -28,3 +30,68 @@ jobs:
     - run: gh pr edit "$PR_NUMBER" --add-label "$(echo "$PR_TITLE" | sed -E 's|([[:alpha:]]+)(\(.*\))?!?:.*|\1|')" || true
       # Extract scope and try to add it as a label
     - run: gh pr edit "$PR_NUMBER" --add-label "$(echo "$PR_TITLE" | sed -E 's|[[:alpha:]]+\((.+)\)!?:.*|\1|')" || true
+
+  add-reviewer:
+    runs-on: ubuntu-latest
+    needs: ["triage", "type-scope"]
+    permissions:
+      contents: write
+      pull-requests: write
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GH_REPO: ${{ github.repository }}
+      PR_NUMBER: ${{ github.event.pull_request.number }}
+
+    steps:
+      - name: Get labels
+        id: labels
+        run: echo "::set-output name=labels::$(gh pr view $PR_NUMBER --json labels --jq '.labels.[].name' | tr '\n' '%')"
+
+        # The % at the end of the label is a hack to avoid selecting a label that
+        # is substring of another label. So if there is a label "cinema" then we
+        # don't accidentally want to interpret that as the label "ci"
+
+      - if: contains(steps.labels.outputs.labels, 'api%')
+        run: gh pr edit $PR_NUMBER --add-reviewer bfredl,gpanders,muniter
+
+      - if: contains(steps.labels.outputs.labels, 'ci%')
+        run: gh pr edit $PR_NUMBER --add-reviewer jamessan
+
+      - if: contains(steps.labels.outputs.labels, 'diagnostic%')
+        run: gh pr edit $PR_NUMBER --add-reviewer gpanders
+
+      - if: contains(steps.labels.outputs.labels, 'distribution%')
+        run: gh pr edit $PR_NUMBER --add-reviewer jamessan
+
+      - if: contains(steps.labels.outputs.labels, 'documentation%')
+        run: gh pr edit $PR_NUMBER --add-reviewer clason
+
+      - if: contains(steps.labels.outputs.labels, 'extmarks%')
+        run: gh pr edit $PR_NUMBER --add-reviewer muniter
```

Comment:
Not really my territory. I only know named marks well :) I guess I can help in the API. I believe most of the work on extmarks has been done by bfredl.